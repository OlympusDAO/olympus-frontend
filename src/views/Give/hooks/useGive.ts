import { t } from "@lingui/macro";
import { ContractReceipt, ethers } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GIVE_ADDRESSES, GOHM_ADDRESSES, SOHM_ADDRESSES } from "src/constants/addresses";
import { IUAData, trackGiveEvent } from "src/helpers/analytics/trackGiveEvent";
import { ACTION_GIVE, getTypeFromAction } from "src/helpers/GiveHelpers";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey } from "src/hooks/useBalance";
import { useDynamicGiveContract } from "src/hooks/useContract";
import { donationInfoQueryKey, recipientInfoQueryKey } from "src/hooks/useGiveInfo";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

import { GiveData } from "../Interfaces";

/**
 * @notice Creates a new deposit
 * @returns ContractReceipt for the deposit
 */
export const useGive = () => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const { address } = useWeb3Context();
  const networks = useTestableNetworks();
  const contract = useDynamicGiveContract(GIVE_ADDRESSES, true);

  // Mutation to interact with the YieldDirector contract
  return useMutation<ContractReceipt, Error, GiveData>(
    // Pass in an object with an amount and a recipient parameter
    async ({ amount: amount_, recipient: recipient_, token: token_ }) => {
      // Validate inputs
      if (parseFloat(amount_) <= 0) throw new Error(t`A give amount must be positive`);

      // Confirm that the user is on a chain where YieldDirector exists
      if (!contract)
        throw new Error(
          t`Give is not supported on this network. Please switch to a supported network, such as Ethereum mainnet`,
        );

      const uaData: IUAData = {
        address: address,
        value: amount_,
        recipient: recipient_,
        approved: true,
        txHash: null,
        type: getTypeFromAction(ACTION_GIVE),
      };

      // Before we submit the transaction, record the event.
      // This lets us track if the user rejects/ignores the confirmation dialog.
      trackGiveEvent(uaData, uaData.type + "-before");

      // Create transaction to deposit passed amount to the passed recipient
      const transaction =
        token_ === "sOHM"
          ? await contract.depositSohm(ethers.utils.parseUnits(amount_, "gwei"), recipient_)
          : await contract.deposit(ethers.utils.parseEther(amount_), recipient_);

      uaData.txHash = transaction.hash;

      trackGiveEvent(uaData);

      return transaction.wait();
    },
    {
      onError: error => {
        console.error(error.message);
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async (data, GiveData) => {
        const keysToRefetch = [
          balanceQueryKey(address, SOHM_ADDRESSES, networks.MAINNET),
          balanceQueryKey(address, GOHM_ADDRESSES, networks.MAINNET),
          donationInfoQueryKey(address, networks.MAINNET),
          recipientInfoQueryKey(GiveData.recipient, networks.MAINNET),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));
        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully deposited sOHM`));
      },
    },
  );
};
