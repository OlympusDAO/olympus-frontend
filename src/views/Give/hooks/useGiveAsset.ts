import { useState } from "react";
import { ChangeAssetType } from "src/slices/interfaces";

export const useGiveAsset = () => {
  const [giveAsset, setGiveAsset] = useState<"sOHM" | "gOHM">("sOHM");

  const changeGiveAsset: ChangeAssetType = (checked: boolean) => {
    setGiveAsset(checked ? "gOHM" : "sOHM");
  };

  return {
    giveAsset,
    changeGiveAsset,
  };
};
