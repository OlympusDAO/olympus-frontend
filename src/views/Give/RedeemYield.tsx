import { t, Trans } from "@lingui/macro";
import { FormControl, Grid, MenuItem, Select, Typography } from "@mui/material";
import { Skeleton } from "@mui/material";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";
import { DataRow, PrimaryButton } from "@olympusdao/component-library";
import { useEffect, useMemo, useState } from "react";
import { GiveBox as Box } from "src/components/GiveProject/GiveBox";
import { DecimalBigNumber } from "src/helpers/DecimalBigNumber/DecimalBigNumber";
import { useCurrentIndex } from "src/hooks/useCurrentIndex";
import { useRecipientInfo, useRedeemableBalance, useV1RedeemableBalance } from "src/hooks/useGiveInfo";
import { useStakingRebaseRate } from "src/hooks/useStakingRebaseRate";
import { useWeb3Context } from "src/hooks/web3Context";
import { GetCorrectContractUnits } from "src/views/Give/helpers/GetCorrectUnits";

import { Project } from "../../components/GiveProject/project.type";
import { GIVE_MAX_DECIMALS } from "./constants";
import { useRedeem } from "./hooks/useRedeem";
import { useOldRedeem } from "./hooks/useRedeemV1";
import data from "./projects.json";
import { RedeemCancelCallback, RedeemYieldModal } from "./RedeemYieldModal";

// Consistent with staking page
const DECIMAL_PLACES = 4;
const ZERO_NUMBER = new DecimalBigNumber("0");
const DECIMAL_FORMAT = { decimals: DECIMAL_PLACES, format: true };
const NO_DECIMAL_FORMAT = { format: true };

export default function RedeemYield() {
  const { address } = useWeb3Context();
  const [isRedeemYieldModalOpen, setIsRedeemYieldModalOpen] = useState(false);
  const theme = useTheme();
  const isSmallScreen = useMediaQuery(theme.breakpoints.down("md"));
  const { projects } = data;
  const projectMap = new Map(projects.map(i => [i.wallet, i] as [string, Project]));
  const [contract, setContract] = useState("new");

  const { data: currentIndex } = useCurrentIndex();

  const _useRedeemableBalance = useRedeemableBalance(address);
  const redeemableBalance: DecimalBigNumber = useMemo(() => {
    if (_useRedeemableBalance.isLoading || _useRedeemableBalance.data === undefined) return new DecimalBigNumber("0");

    return GetCorrectContractUnits(_useRedeemableBalance.data, "gOHM", currentIndex);
  }, [_useRedeemableBalance, currentIndex]);

  const _useV1RedeemableBalance = useV1RedeemableBalance(address);
  const v1RedeemableBalance: DecimalBigNumber = useMemo(() => {
    if (_useV1RedeemableBalance.isLoading || _useV1RedeemableBalance.data === undefined)
      return new DecimalBigNumber("0");

    return new DecimalBigNumber(_useV1RedeemableBalance.data, 9);
  }, [_useV1RedeemableBalance]);

  const _useRecipientInfo = useRecipientInfo(address);
  const isRecipientInfoLoading = _useRecipientInfo.isLoading;

  const _useStakingRebaseRate = useStakingRebaseRate();
  const isStakingRebaseRateLoading = _useStakingRebaseRate.isLoading;
  const stakingRebase: DecimalBigNumber = useMemo(() => {
    if (_useStakingRebaseRate.isLoading || _useStakingRebaseRate.data === undefined) return ZERO_NUMBER;

    return new DecimalBigNumber(_useStakingRebaseRate.data.toString());
  }, [_useStakingRebaseRate]);

  const fiveDayRate: DecimalBigNumber = useMemo(() => {
    if (stakingRebase.eq(ZERO_NUMBER)) return ZERO_NUMBER;

    return new DecimalBigNumber((Math.pow(1 + stakingRebase.toApproxNumber(), 5 * 3) - 1).toString());
  }, [stakingRebase]);

  const totalDebt: DecimalBigNumber = useMemo(() => {
    if (_useRecipientInfo.isLoading || _useRecipientInfo.data == undefined) return ZERO_NUMBER;

    return new DecimalBigNumber(_useRecipientInfo.data.sohmDebt);
  }, [_useRecipientInfo]);

  const stakingRebasePercentage = stakingRebase.mul(new DecimalBigNumber("100"));

  const nextRewardValue = stakingRebase.mul(totalDebt);

  const fiveDayRateValue = fiveDayRate.mul(new DecimalBigNumber("100"));

  const isProject = projectMap.get(address);

  const redeemMutation = useRedeem();
  const isMutating = redeemMutation.isLoading;

  const oldRedeemMutation = useOldRedeem();

  useEffect(() => {
    if (isRedeemYieldModalOpen) setIsRedeemYieldModalOpen(false);
  }, [redeemMutation.isSuccess, oldRedeemMutation.isSuccess]);

  // Get project sOHM yield goal and return as a number
  const getRecipientGoal = (address: string): DecimalBigNumber => {
    const project = projectMap.get(address);
    if (project) return new DecimalBigNumber(project.depositGoal.toString());

    return ZERO_NUMBER;
  };

  const getRedeemableBalance = (): DecimalBigNumber => {
    return contract === "new" ? redeemableBalance : v1RedeemableBalance;
  };

  /**
   * Checks that the current user can redeem some quantity of sOHM
   *
   * @returns
   */
  const canRedeem = () => {
    if (!address) return false;

    if (isRecipientInfoLoading) return false;

    if (isMutating) return false;

    if (contract === "new") {
      if (redeemableBalance.eq(ZERO_NUMBER)) return false;
    } else {
      if (v1RedeemableBalance.eq(ZERO_NUMBER)) return false;
    }

    return true;
  };

  const handleRedeemButtonClick = () => {
    setIsRedeemYieldModalOpen(true);
  };

  const handleRedeemYieldModalSubmit = async () => {
    await redeemMutation.mutate({ token: "sOHM" });
  };

  const handleOldRedeemYieldModalSubmit = async () => {
    await oldRedeemMutation.mutate();
  };

  const handleRedeemYieldModalCancel: RedeemCancelCallback = () => {
    setIsRedeemYieldModalOpen(false);
  };

  return (
    <Grid container spacing={2}>
      {v1RedeemableBalance.gt(ZERO_NUMBER) && (
        <Grid container justifyContent="flex-end">
          <Box overrideClass="redeem-selector">
            <Grid item xs={12}>
              <FormControl>
                <Select
                  label="Contract"
                  disableUnderline
                  id="contract-select"
                  value={contract === "new" ? "new" : "old"}
                  onChange={event => setContract(event.target.value === "new" ? "new" : "old")}
                >
                  <MenuItem value="new">Redeem from new contract</MenuItem>
                  <MenuItem value="old">Redeem from old contract</MenuItem>
                </Select>
              </FormControl>
            </Grid>
          </Box>
        </Grid>
      )}
      <Grid item xs={12}>
        <Typography variant="h3" align="center" data-testid="redeemable-balance">
          {isRecipientInfoLoading ? <Skeleton /> : getRedeemableBalance().toString(DECIMAL_FORMAT)} sOHM
        </Typography>
        <Typography variant="body1" align="center" className="subtext">
          Redeemable Yield
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Grid container>
          <Grid item xs />
          <Grid item xs={12} sm={6}>
            <PrimaryButton onClick={() => handleRedeemButtonClick()} disabled={!canRedeem()} fullWidth>
              <Trans>Redeem Yield</Trans>
            </PrimaryButton>
          </Grid>
          <Grid item xs />
        </Grid>
      </Grid>
      {isProject ? (
        <Grid item xs={12}>
          <Grid container spacing={1}>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center">
                  {getRecipientGoal(address).toString(DECIMAL_FORMAT)}
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  <Trans>sOHM Goal</Trans>
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center" data-testid="project-deposit">
                  {totalDebt.toString(DECIMAL_FORMAT)}
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  {isSmallScreen ? "Total Donated" : `Total sOHM Donated`}
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={4}>
              <Box>
                <Typography variant="h5" align="center" data-testid="project-goal-achievement">
                  {totalDebt
                    .mul(new DecimalBigNumber("100"))
                    .div(getRecipientGoal(address), GIVE_MAX_DECIMALS)
                    .toString(DECIMAL_FORMAT)}
                  %
                </Typography>
                <Typography variant="body1" align="center" className="subtext">
                  <Trans>of sOHM Goal</Trans>
                </Typography>
              </Box>
            </Grid>
          </Grid>
        </Grid>
      ) : (
        <></>
      )}
      <Grid item xs={12}>
        <Box>
          <DataRow
            title={t`Deposited sOHM`}
            // Exact number
            balance={`${contract === "new" ? totalDebt.toString(NO_DECIMAL_FORMAT) : ZERO_NUMBER} ${t`sOHM`}`}
            isLoading={isRecipientInfoLoading}
            data-testid="data-deposited-sohm"
          />
          <DataRow
            title={t`Redeemable Amount`}
            balance={`${getRedeemableBalance().toString(NO_DECIMAL_FORMAT)} ${t`sOHM`}`}
            isLoading={isRecipientInfoLoading}
            data-testid="data-redeemable-balance"
          />
          <DataRow
            title={t`Next Reward Amount`}
            balance={`${nextRewardValue.toString(DECIMAL_FORMAT)} ${t`sOHM`}`}
            isLoading={isStakingRebaseRateLoading}
            data-testid="data-next-reward-amount"
          />
          <DataRow
            title={t`Next Reward Yield`}
            balance={`${stakingRebasePercentage.toString(DECIMAL_FORMAT)}%`}
            isLoading={isStakingRebaseRateLoading}
            data-testid="data-next-reward-yield"
          />
          <DataRow
            title={t`ROI (5-Day Rate)`}
            balance={`${fiveDayRateValue.toString(DECIMAL_FORMAT)}%`}
            isLoading={isStakingRebaseRateLoading}
            data-testid="data-roi"
          />
        </Box>
      </Grid>
      <Grid item>
        <RedeemYieldModal
          isModalOpen={isRedeemYieldModalOpen}
          callbackFunc={contract === "new" ? handleRedeemYieldModalSubmit : handleOldRedeemYieldModalSubmit}
          cancelFunc={handleRedeemYieldModalCancel}
          contract={contract}
          deposit={totalDebt}
          redeemableBalance={getRedeemableBalance()}
          isMutationLoading={isMutating}
        />
      </Grid>
    </Grid>
  );
}
