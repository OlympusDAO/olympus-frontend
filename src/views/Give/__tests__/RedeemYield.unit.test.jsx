import { configureStore } from "@reduxjs/toolkit";
import * as useWeb3Context from "src/hooks/web3Context";
import accountReducer from "src/slices/AccountSlice";
import appReducer from "src/slices/AppSlice";
import pendingTransactionsReducer from "src/slices/PendingTxnsSlice";
import * as Pending from "src/slices/PendingTxnsSlice";
import { mockWeb3Context } from "src/testHelpers";

import { act, render, screen } from "../../../testUtils";
import RedeemYield from "../RedeemYield";

let store;
let redeemingStore;
let context;
beforeEach(() => {
  context = jest.spyOn(useWeb3Context, "useWeb3Context");
  const preloadedState = {
    app: {
      loading: false,
    },
    account: {
      giving: {
        sohmGive: 999999999000000000,
        donationInfo: [
          {
            date: "Mar 30, 2022",
            deposit: "1.0",
            recipient: "0xd3B4a9604c78DDA8692d85Dc15802BA12Fb82b6c",
            yieldDonated: "0.0",
          },
        ],
        loading: false,
      },
    },
  };

  const redeemingState = {
    app: {
      circSupply: 15203090.685576908,
      currentBlock: 14523245,
      currentIndex: "112.391236227",
      currentIndexV1: "46.721314322",
      fiveDayRate: 0.03269220242927484,
      loading: false,
      loadingMarketPrice: false,
      marketCap: 500132858.5349252,
      marketPrice: 32.90090902231928,
      secondsToEpoch: 5812,
      stakingAPY: 9.46828503212633,
      stakingRebase: 0.002146913392161418,
      stakingTVL: 405991286.3415339,
      totalSupply: 16732992.484142939,
      treasuryMarketValue: 444014377.20682436,
    },
    account: {
      giving: {
        sohmGive: 999999999000000000,
        donationInfo: [
          {
            date: "Mar 30, 2022",
            deposit: "1.0",
            recipient: "0xd3B4a9604c78DDA8692d85Dc15802BA12Fb82b6c",
            yieldDonated: "0.0",
          },
        ],
        loading: false,
      },
      redeeming: {
        sohmRedeemable: "100",
        recipientInfo: {
          totalDebt: "100.0",
          carry: "0.0",
          agnosticDebt: "0.0",
          indexAtLastChange: "0.0",
        },
      },
    },
  };

  const reducer = {
    account: accountReducer,
    app: appReducer,
    pendingTransactions: pendingTransactionsReducer,
  };

  store = configureStore({
    reducer,
    devTools: true,
    preloadedState,
  });

  redeemingStore = configureStore({
    reducer,
    devTools: true,
    preloadedState: redeemingState,
  });
});

afterEach(() => {
  //jest.restoreAllMocks();
  jest.clearAllMocks();
});

describe("Redeem Yield", () => {
  it("should render Redeem Yield Screen", async () => {
    context.mockReturnValue(mockWeb3Context);
    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />));
    });
    expect(container).toMatchSnapshot();
  });

  it("should have disabled redeem button when there are pending transaction(s)", async () => {
    context.mockReturnValue(mockWeb3Context);
    const pending = jest.spyOn(Pending, "isPendingTxn");
    pending.mockReturnValue(true);
    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />, store)); //eslint-disable-line
    });
    expect(screen.getByText("Redeem Yield").closest("button")).toHaveAttribute("disabled");
  });

  it("should show total deposit as 100 sOHM", async () => {
    context.mockReturnValue(mockWeb3Context);
    const pending = jest.spyOn(Pending, "isPendingTxn");
    pending.mockReturnValue(true);
    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />, redeemingStore)); //eslint-disable-line
    });
    expect(container).toMatchSnapshot();
    expect(screen.getByText("100.00 sOHM")).toBeInTheDocument();
  });

  it("should show extra content if project wallet", async () => {
    context.mockReturnValue({ ...mockWeb3Context, address: "0xd3B4a9604c78DDA8692d85Dc15802BA12Fb82b6c" });
    let container;
    await act(async () => {
      ({ container } = render(<RedeemYield />, redeemingStore)); //eslint-disable-line
    });
    expect(screen.getByText("sOHM Goal")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
