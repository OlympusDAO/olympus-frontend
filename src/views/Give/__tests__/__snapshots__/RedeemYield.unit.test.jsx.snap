// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redeem Yield should render Redeem Yield Screen 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        />
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 makeStyles-root-2 undefined MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
        <div
          class="makeStyles-root-4 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-10"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;

exports[`Redeem Yield should show extra content if project wallet 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
      >
        100.00
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-40 makeStyles-root-41 undefined MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-42"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              200
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              sOHM Goal
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-43"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              0
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              Total sOHM Donated
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-44"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              0
              %
            </h5>
            <p
              class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
            >
              of sOHM Goal
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-45"
      >
        <div
          class="makeStyles-root-46 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-48"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            100 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-46 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-49"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            100 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-46 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-50"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0.2147 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-46 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-51"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0.2147%
          </p>
        </div>
        <div
          class="makeStyles-root-46 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-52"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            3.2692%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;

exports[`Redeem Yield should show total deposit as 100 sOHM 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
      >
        100.00
         sOHM
      </h3>
      <p
        class="MuiTypography-root subtext MuiTypography-body1 MuiTypography-alignCenter"
      >
        Redeemable Yield
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-27 makeStyles-root-28 undefined MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Redeem Yield
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-29"
      >
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-32"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Deposited sOHM
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            100 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-33"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Redeemable Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            100 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-34"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Amount
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0.2147 sOHM
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-35"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              Next Reward Yield
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            0.2147%
          </p>
        </div>
        <div
          class="makeStyles-root-30 data-row"
        >
          <div
            class="MuiBox-root MuiBox-root-36"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            >
              ROI (5-Day Rate)
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
            style="text-align: right;"
          >
            3.2692%
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;
